### ISSUE TITLE
Feature request: Nextjs on EC2 or Cluster (Docker)

### ISSUE THREAD

**Issue by agamm** [anomalyco/sst#5206]
Right now, we have to use lambda for the server, it would be awesome if we could use Next like so:

```js
new sst.aws.Nextjs("MyWeb", {
  path: "my-next-app/"
  server: {
    runtime: // lambda / ec2 / docker
  }
});
```

Using [ai sdk](https://sdk.vercel.ai/docs/introduction) I am getting a lot of problems streaming data back to the client (guessing because of lambda's not supposed to) - If I could just run next easily on a long running server that would be awesome, and also probably remove the need for things like trigger[dot]dev or inngest.

**Comment by jayair** [comment]
Any reason you don't want to run it fully in a container? https://sst.dev/docs/start/aws/nextjs#containers

**Comment by agamm** [comment]
I'm still trying to get Cluster to work, I have a pnpm workspace and the dockerfile is giving me trouble. I'll eventually get it working but having it work out of the box would be great. I'd also love to use the sst console with it so I'll have more observability. 

(EC2 is just there to indicate a server and not serverless lambda,docker is also good).

**Comment by agamm** [comment]
@jayair For example i'm following the link you added, and I already wasted half a day tring to fiddle with docker to make the build pass, currently stuck with `It does not look like SST links are active.` during my build. Having it work ootb would be much appreciated.

**Comment by agamm** [comment]
@jayair any update on making this work? aws.Next is getting me really flaky results when my nextjs server actions run for a long time (3+ minutes) - I want to be able to run long running tasks from next and cluster seems like the only answer...

**Comment by jayair** [comment]
Yeah sorry did you mean supporting running the Nextjs component in a cluster? I'm not sure if that's on our short/medium term roadmap.

But we do have a lot of folks running Next.js in a Cluster component. And I think I had responded to your other issue here https://github.com/sst/sst/issues/5227

**Comment by agamm** [comment]
Ok, could we keep this open until there is a more easy solution to run nextjs than manually setting a cluster up? I wasted a few days trying to make it work with secrets so, I'm sure a few others would still like an easy way to run nextjs not on lambdas for long running tasks.

**Comment by thdxr** [comment]
we have all this documented now
